src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```diff
@@ -99,16 +105,55 @@ public boolean isJava8() {
             && getRuntime().startsWith("java8");
   }
 
-  private static String getTopLevelValue(Document doc, String parentTagName, String childTagName) {
+  public Map<String, String> getEnvironment() {
+    Node environmentParentNode = getTargetNode(document, "appengine-web-app", "env-variables");
+    return getAttributeMap(environmentParentNode, "name", "value");
+  }
+
+  private static String getText(Node node) {
+    if (node != null) {
+      return node.getTextContent();
+    }
+
+    return null;
+  }
+
+  private static Map<String, String> getAttributeMap(Node node, String key, String value) {
```

Comment:
I realized I need to update this to also search for a target child node by name. E.g.:

```
<env-variables>
        <env-var name="ENDPOINTS_SERVICE_NAME" value="some_value" />
 </env-variables>
```
The logic should probably search for the tag 'env-var'.